### ISSUE TITLE
Manuel installation error : jobs.py : OverflowError: Python int too large to convert to SQLite INTEGER

### ISSUE THREAD

**Issue by hoplaaa** [cryptosharks131/lndg#38]
Following : https://github.com/cryptosharks131/lndg#step-1---install-lndg

I get an error at step 8. Generate some initial data for your dashboard .venv/bin/python jobs.py 

Tried to delete the db and redo the initialization.
Tried to remove lndg folder and restart from step 1.

```
user@hostname:~$ git clone https://github.com/cryptosharks131/lndg.git
Cloning into 'lndg'...
remote: Enumerating objects: 2108, done.
remote: Counting objects: 100% (2085/2085), done.
remote: Compressing objects: 100% (878/878), done.
remote: Total 2108 (delta 1469), reused 1704 (delta 1149), pack-reused 23
Receiving objects: 100% (2108/2108), 686.20 KiB | 2.48 MiB/s, done.
Resolving deltas: 100% (1469/1469), done.
user@hostname:~$ cd lndg

user@hostname:~/lndg$ virtualenv -p python3 .venv
created virtual environment CPython3.9.2.final.0-64 in 216ms
  creator CPython3Posix(dest=/home/user/lndg/.venv, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/user/.local/share/virtualenv)
    added seed packages: pip==20.3.4, pkg_resources==0.0.0, setuptools==44.1.1, wheel==0.34.2
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator

user@hostname:~/lndg$ .venv/bin/pip install -r requirements.txt
Collecting Django
  Using cached Django-4.0.2-py3-none-any.whl (8.0 MB)
Collecting djangorestframework
  Using cached djangorestframework-3.13.1-py3-none-any.whl (958 kB)
Collecting django-qr-code
  Using cached django_qr_code-3.0.0-py3-none-any.whl (28 kB)
Collecting grpcio
  Using cached grpcio-1.43.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.1 MB)
Collecting protobuf
  Using cached protobuf-3.19.4-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.1 MB)
Collecting pytz
  Using cached pytz-2021.3-py2.py3-none-any.whl (503 kB)
Collecting pandas
  Using cached pandas-1.4.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.7 MB)
Collecting sqlparse>=0.2.2
  Using cached sqlparse-0.4.2-py3-none-any.whl (42 kB)
Collecting asgiref<4,>=3.4.1
  Using cached asgiref-3.5.0-py3-none-any.whl (22 kB)
Collecting segno
  Using cached segno-1.4.1-py2.py3-none-any.whl (82 kB)
Collecting six>=1.5.2
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting numpy>=1.18.5
  Using cached numpy-1.22.2-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.8 MB)
Collecting python-dateutil>=2.8.1
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Installing collected packages: sqlparse, six, asgiref, segno, pytz, python-dateutil, numpy, Django, protobuf, pandas, grpcio, djangorestframework, django-qr-code
Successfully installed Django-4.0.2 asgiref-3.5.0 django-qr-code-3.0.0 djangorestframework-3.13.1 grpcio-1.43.0 numpy-1.22.2 pandas-1.4.0 protobuf-3.19.4 python-dateutil-2.8.2 pytz-2021.3 segno-1.4.1 six-1.16.0 sqlparse-0.4.2

user@hostname:~/lndg$ .venv/bin/python initialize.py
Setting up initial user...
Superuser created successfully.
FIRST TIME LOGIN PASSWORD:

user@hostname:~/lndg$ .venv/bin/python jobs.py
Traceback (most recent call last):
  File "/home/user/lndg/jobs.py", line 37, in update_payments
    Paymenthostnames(payment_hash=new_payment, attempt_id=attempt.attempt_id, step=hostname_count, chan_id=hostname.chan_id, alias=alias, chan_capacity=hostname.chan_capacity, node_pubkey=hostname.pub_key, amt=round(hostname.amt_to_forward_msat/1000, 3), fee=round(fee, 3), cost_to=round(cost_to, 3)).save()
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/models/base.py", line 743, in save
    self.save_base(using=using, force_insert=force_insert,
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/models/base.py", line 780, in save_base
    updated = self._save_table(
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/models/base.py", line 885, in _save_table
    results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/models/base.py", line 923, in _do_insert
    return manager._insert(
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/models/query.py", line 1301, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/models/sql/compiler.py", line 1441, in execute_sql
    cursor.execute(sql, params)
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/backends/utils.py", line 76, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/backends/sqlite3/base.py", line 416, in execute
    return Database.Cursor.execute(self, query, params)
OverflowError: Python int too large to convert to SQLite INTEGER

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/user/lndg/jobs.py", line 248, in <module>
    main()
  File "/home/user/lndg/jobs.py", line 241, in main
    update_payments(stub)
  File "/home/user/lndg/jobs.py", line 74, in update_payments
    Paymenthostnames(payment_hash=db_payment, attempt_id=attempt.attempt_id, step=hostname_count, chan_id=hostname.chan_id, alias=alias, chan_capacity=hostname.chan_capacity, node_pubkey=hostname.pub_key, amt=round(hostname.amt_to_forward_msat/1000, 3), fee=round(fee, 3), cost_to=round(cost_to, 3)).save()
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/models/base.py", line 743, in save
    self.save_base(using=using, force_insert=force_insert,
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/models/base.py", line 780, in save_base
    updated = self._save_table(
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/models/base.py", line 885, in _save_table
    results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/models/base.py", line 923, in _do_insert
    return manager._insert(
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/models/query.py", line 1301, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/models/sql/compiler.py", line 1441, in execute_sql
    cursor.execute(sql, params)
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/backends/utils.py", line 76, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
  File "/home/user/lndg/.venv/lib/python3.9/site-packages/django/db/backends/sqlite3/base.py", line 416, in execute
    return Database.Cursor.execute(self, query, params)
OverflowError: Python int too large to convert to SQLite INTEGER
```

**Comment by cryptosharks131** [comment]
This is very strange, not sure how to replicate this one as it is something specific to your node's dataset.

It is trying to insert a value into the database and failing due to a variable out of bounds.

Would you be able to update this section of code and report back if the output you get looks to have any issues? It will print out the 3 values that are trying to be inserted as an integer.

Here is the line that needs to be changed.
https://github.com/cryptosharks131/lndg/blob/master/jobs.py#L74

Update this single line to the following, making sure to respect the existing indentations.
```
try:
    PaymentHops(payment_hash=new_payment, attempt_id=attempt.attempt_id, step=hop_count, chan_id=hop.chan_id, alias=alias, chan_capacity=hop.chan_capacity, node_pubkey=hop.pub_key, amt=round(hop.amt_to_forward_msat/1000, 3), fee=round(fee, 3), cost_to=round(cost_to, 3)).save()
except Exception as e:
    print('Caught Error: ' + str(e))
    print('Channel ID: ' + hop.chan_id)
    print('Attempt ID: ' + attempt.attempt_id)
    print('Step: ' + hop_count)
```

**Comment by hoplaaa** [comment]
I added str() on the 3 lasts line after some concatenation issue.
```

try:
    PaymentHops(payment_hash=new_payment, attempt_id=attempt.attempt_id, step=hop_count, chan_id=hop.chan_id, alias=alias, chan_capacity=hop.chan_capacity, node_pubkey=hop.pub_key, amt=round(hop.amt_to_forward_msat/1000, 3), fee=round(fee, 3), cost_to=round(cost_to, 3)).save()
except Exception as e:
    print('Caught Error: ' + str(e))
    print('Channel ID: ' + str(hop.chan_id))
    print('Attempt ID: ' + str(attempt.attempt_id))
    print('Step: ' + str(hop_count))
```

Here is the result :
```
:~/lndg$ .venv/bin/python jobs.py
Caught Error: Python int too large to convert to SQLite INTEGER
Channel ID: 18263241250478212038
Attempt ID: 2002
Step: 2
Caught Error: Python int too large to convert to SQLite INTEGER
Channel ID: 9420252987517611809
Attempt ID: 2092
Step: 2
Caught Error: Python int too large to convert to SQLite INTEGER
Channel ID: 18320464923357465051
Attempt ID: 2251
Step: 2
Caught Error: Python int too large to convert to SQLite INTEGER
Channel ID: 9890057090840971557
Attempt ID: 6000
Step: 3
Caught Error: Python int too large to convert to SQLite INTEGER
Channel ID: 15505268299679684212
Attempt ID: 6001
Step: 3
```

These IDs are not matchgin my channels, they have 2 more digit.

**Comment by cryptosharks131** [comment]
Good catch with the string issue.  Can you also print out the payment id and index?

```
print('Payment Hash: ' + str(payment.payment_hash))
print('Index: ' + str(payment.payment_index))
```

Then use lncli to see if this same data can be pulled directly from LND. You can use the index value above to directly look the payment up with lnci.

```
lncli listpayments --index_offset <offset_value> --max_payments 1
```

My thought would be that if its also showing an incorrect channel ID directly from lncli then the root cause might be a bug in LND itself.

**Comment by hoplaaa** [comment]
Added the 2 lines, reran jobs.py: no output, no more error, went to the next step without issue.

I have lnd log in debug mode, tried to grep these numbers there, no success.

**Comment by hoplaaa** [comment]
To answer the 2nd part of your request, here is one output for the last offset :

lncli listpayments --index_offset 15505268299679684212 --max_payments 1
{
    "payments": [
        {
            "payment_hash": "removed",
            "value": "4000000",
            "creation_date": "1643974133",
            "fee": "0",
            "payment_preimage": "removed",
            "value_sat": "4000000",
            "value_msat": "4000000000",
            "payment_request": "",
            "status": "SUCCEEDED",
            "fee_sat": "0",
            "fee_msat": "0",
            "creation_time_ns": "1643974133409064929",
            "htlcs": [
                {
                    "attempt_id": "15150",
                    "status": "SUCCEEDED",
                    "route": {
                        "total_time_lock": 721831,
                        "total_fees": "0",
                        "total_amt": "4000000",
                        "hops": [
                            {
                                "chan_id": "removed",
                                "chan_capacity": "5000000",
                                "amt_to_forward": "4000000",
                                "fee": "0",
                                "expiry": 721831,
                                "amt_to_forward_msat": "4000000000",
                                "fee_msat": "0",
                                "pub_key": "removed",
                                "tlv_payload": true,
                                "mpp_record": {
                                    "payment_addr": "removed",
                                    "total_amt_msat": "4000000000"
                                },
                                "amp_record": null,
                                "custom_records": {
                                }
                            }
                        ],
                        "total_fees_msat": "0",
                        "total_amt_msat": "4000000000"
                    },
                    "attempt_time_ns": "1643974133428292463",
                    "resolve_time_ns": "1643974134664430983",
                    "failure": null,
                    "preimage": "removed"
                }
            ],
            "payment_index": "13169",
            "failure_reason": "FAILURE_REASON_NONE"
        }
    ],
    "first_index_offset": "13169",
    "last_index_offset": "13169"
}

**Comment by cryptosharks131** [comment]
Very interesting, not sure what might have changed here but good that its working now.

I noticed the <index_offset> you used was a bit off. It should be the value returned from this:
```
print('Index: ' + str(payment.payment_index))
```

That will match what you see from the lncli output under `payment_index` if its indeed the same payment (payment hash will also match).

For example. the output you gave above has a payment index of `13169`.

**Comment by hoplaaa** [comment]
I see, my previous message has no value. Sorry for that.
Also, I cannot reproduce the issue to get the correct `payment_index`.

I could run `lncli listpayments > file` and searched there, found the payment and the channel id. 

Payments correspond to payments I made to muun, last hop between magnetron and the muun wallet is the channel is long id.

The problem is that SQLite expect a signed integer, but botlz and lnd can use the entire signed range.
Bolts : https://github.com/lightning/bolts/search?q=short_channel_id
`short_channel_id (64 bits)`

LND : https://github.com/lightningnetwork/lnd/search?q=chan_id
`"format": "uint64"`

SQLite : https://www.sqlite.org/datatype3.html
`"INTEGER. The value is a signed integer, stored in 0, 1, 2, 3, 4, 6, or 8 bytes depending on the magnitude of the value."`

`18446744073709551615 ` : max unsigned 8byte (64bit) int
`15505268299679684212 `<- OverflowError, int too large for sqllite
`9223372036854775807 ` : max signed 8byte (64bit) int

You can find bellow the example for the last, `15505268299679684212`.

```
       {
            "payment_hash": "removed",
            "value": "300000",
            "creation_date": "1639069085",
            "fee": "480",
            "payment_preimage": "removed",
            "value_sat": "300000",
            "value_msat": "300000000",
            "payment_request": "removed",
            "status": "SUCCEEDED",
            "fee_sat": "480",
            "fee_msat": "480145",
            "creation_time_ns": "1639069085451384845",
            "htlcs": [
                {
                    "attempt_id": "6001",
                    "status": "SUCCEEDED",
                    "route": {
                        "total_time_lock": 713521,
                        "total_fees": "480",
                        "total_amt": "300480",
                        "hops": [
                            {
                                "chan_id": "removed",
                                "chan_capacity": "1000000",
                                "amt_to_forward": "300450",
                                "fee": "30",
                                "expiry": 713481,
                                "amt_to_forward_msat": "300450100",
                                "fee_msat": "30045",
                                "pub_key": "removed",
                                "tlv_payload": true,
                                "mpp_record": null,
                                "amp_record": null,
                                "custom_records": {
                                }
                            },
                            {
                                "chan_id": "783825346926542849",
                                "chan_capacity": "300450",
                                "amt_to_forward": "300000",
                                "fee": "450",
                                "expiry": 713472,
                                "amt_to_forward_msat": "300000000",
                                "fee_msat": "450100",
                                "pub_key": "038f8f113c580048d847d6949371726653e02b928196bad310e3eda39ff61723f6",
                                "tlv_payload": true,
                                "mpp_record": null,
                                "amp_record": null,
                                "custom_records": {
                                }
                            },
                            {
                                "chan_id": "15505268299679684212",
                                "chan_capacity": "300000",
                                "amt_to_forward": "300000",
                                "fee": "0",
                                "expiry": 713472,
                                "amt_to_forward_msat": "300000000",
                                "fee_msat": "0",
                                "pub_key": "removed",
                                "tlv_payload": true,
                                "mpp_record": {
                                    "payment_addr": "removed",
                                    "total_amt_msat": "300000000"
                                },
                                "amp_record": null,
                                "custom_records": {
                                }
                            }
                        ],
                        "total_fees_msat": "480145",
                        "total_amt_msat": "300480145"
                    },
                    "attempt_time_ns": "1639069085518435139",
                    "resolve_time_ns": "1639069090993892080",
                    "failure": null,
                    "preimage": "removed"
                }
            ],
            "payment_index": "4010",
            "failure_reason": "FAILURE_REASON_NONE"
        },

```

**Comment by cryptosharks131** [comment]
Thank you for all of this detail, it seems like you have found the issue!

Since this appears to be a valid channel id, it may be better to start storing these channel ids as a string similar to the pubkey is currently but with a max length of 20.

**Comment by cryptosharks131** [comment]
Resolved in #51
